crates/autopilot/src/infra/blockchain/contracts.rs

```diff
@@ -20,7 +20,7 @@ pub struct Addresses {
 }
 
 impl Contracts {
-    pub async fn new(web3: &DynWeb3, chain: &ChainId, addresses: Addresses) -> Self {
+    pub async fn new(web3: &DynWeb3, chain: &domain::eth::chain::Id, addresses: Addresses) -> Self {
```

Comment:
nit: `domain::eth::chain` seems weird.
A structure like this make more sense to me:
chain
----> id
----> mod

Reply:
Hm, not following, how would the path go in your case?

Reply:
`domain::chain::Id` instead of `domain::eth::chain::Id`.
It doesn't make much sense to me to have the less specific `chain` as a child of the more specific `eth` module.

Reply:
Hm, to me chain is not that important concept to live directly under `domain`. Previously it lived under `infra::blockchain`, maybe we should move it there, so we would have `infra::blockchain::Id`?

Reply:
Moved to infra, probably better now